tiles-generation/cdk/bin/cdk.ts

```diff
@@ -11,110 +11,167 @@ import { App } from 'aws-cdk-lib';
 import { TileGenerationStack } from '../lib/tile-generation-stack';
 import { SlackNotificationStack } from '../lib/slack-notification-stack';
 import { EmailNotificationStack } from '../lib/email-notification-stack';
-import * as ec2 from "aws-cdk-lib/aws-ec2";
-import * as asc from "aws-cdk-lib/aws-autoscaling";
+import * as ec2 from 'aws-cdk-lib/aws-ec2';
+import * as asc from 'aws-cdk-lib/aws-autoscaling';
 
 const app = new App();
 
 const env = {
-  region: app.node.tryGetContext('region') || process.env.CDK_INTEG_REGION || process.env.CDK_DEFAULT_REGION,
-  account: app.node.tryGetContext('account') || process.env.CDK_INTEG_ACCOUNT || process.env.CDK_DEFAULT_ACCOUNT
+    region:
+        app.node.tryGetContext('region') ||
+        process.env.CDK_INTEG_REGION ||
+        process.env.CDK_DEFAULT_REGION,
+    account:
+        app.node.tryGetContext('account') ||
+        process.env.CDK_INTEG_ACCOUNT ||
+        process.env.CDK_DEFAULT_ACCOUNT,
 };
 
-const testInstanceType = ec2.InstanceType.of(ec2.InstanceClass.MEMORY3, ec2.InstanceSize.XLARGE);
+const testInstanceType = ec2.InstanceType.of(
```

Comment:
what is the meaning of testinstance here? rename this variable

Reply:
Okay understood make sense, we should test the setup. But in terms of code, I can see a lot duplication and no clear separation between the two stack, which make difficult to understand.

I would recommend creating 2 new private function(createSetupValidationStack, createProdStack) one for the test stack and one for actual stack.

Provide the details on both the functions what they are doing and why they are necessary. 

Also abstract out the common logic between the validation stack and prod stack.

Reply:
Thanks! I just updated the code.